File: addCategories.mjs
import fs from 'fs';

const categories = [
  { name: 'Shoes', keywords: ['shoe', 'sneaker', 'trainer', 'yeezy', 'air force', 'jordan', 'dunk', 'asics', 'adidas', 'puma', 'balenciaga', 'mcqueen', 'new balance', 'golden goose', 'nike'] },
  { name: 'Jackets', keywords: ['jacket', 'coat', 'parka', 'windbreaker'] },
  { name: 'Hoodies/Sweaters', keywords: ['hoodie', 'sweater', 'crewneck', 'pullover'] },
  { name: 'T-shirts', keywords: ['t-shirt', 'tee', 'shirt'] },
  { name: 'Pants', keywords: ['pant', 'jean', 'trouser', 'cargo', 'track pant', 'short'] },
  { name: 'Hats', keywords: ['hat', 'cap', 'beanie', 'bucket'] },
  { name: 'Accessories', keywords: ['bag', 'belt', 'bracelet', 'watch', 'wallet', 'ring', 'necklace', 'sunglass', 'accessory'] },
  { name: 'Sportswear', keywords: ['sport', 'jersey', 'gym', 'athletic', 'tracksuit'] }
];

const products = JSON.parse(fs.readFileSync('public/products_with_images.json', 'utf8'));

function getCategory(title) {
  const lower = title.toLowerCase();
  for (const cat of categories) {
    if (cat.keywords.some(k => lower.includes(k))) {
      return cat.name;
    }
  }
  return 'Other';
}

const updated = products.map(p => ({
  ...p,
  category: getCategory(p[""] || "")
}));

fs.writeFileSync('public/products_with_images.json', JSON.stringify(updated, null, 2));
console.log('✅ Categories added!');

File: addCategories.py
import json

# Map keywords to your categories
category_map = {
    "Shoes": "Shoes",
    "Jacket": "Jackets",
    "Hoodie": "Hoodies/Sweaters",
    "Sweater": "Hoodies/Sweaters",
    "Sweatshirt": "Hoodies/Sweaters",
    "T-shirt": "T-shirts",
    "Tee": "T-shirts",
    "Pants": "Pants",
    "Trousers": "Pants",
    "Jeans": "Pants",
    "Shorts": "Pants",
    "Hat": "Hats",
    "Cap": "Hats",
    "Baseball": "Hats",
    "Accessories": "Accessories",
    "Sunglasses": "Accessories",
    "Necklace": "Accessories",
    "Mask": "Accessories",
    "Perfume": "Accessories",
    "Chain": "Accessories",
}

def get_category(title):
    for keyword, category in category_map.items():
        if keyword.lower() in title.lower():
            return category
    return "Other"

with open('products_with_images_scraped.json', 'r', encoding='utf-8') as f:
    products = json.load(f)

for product in products:
    product['category'] = get_category(product['title'])

with open('products_with_images_scraped.json', 'w', encoding='utf-8') as f:
    json.dump(products, f, indent=2, ensure_ascii=False)